### DISCUSSION TITLE
Cable connection between a single 1G SFP port on a device to 2 Front ports on patch panel (one Tx and one Rx)

### DISCUSSION THREAD

**Discussion by subhashkoganti** [netbox-community/netbox#9089] [Q&A] [answered]
Hi,

This is my first time in this netbox community.

I need to create a cable connection between:

a single 1G SFP port interface on a device 
and 
2 Front ports on patch panel (one Tx and one Rx)

My Observation:
I able to connect 1 Interface on a Device to only 1 Front Port or 1 Rear Port or 1 Interface.
I don't see any possibility on the netbox UI to be able to connect 1 device Interface with 2 Front ports on a patch panel (one for Rx and one for Tx). I even tried changing Front port types on the patch panel, but with no luck.

My question:
Is this even possible? If yes, what am I missing.
If no, is there a pending feature ticket, that will address this.

Thanks in advance

regards,
Subhash

**Comment by candlerb** [comment, marked as answer]
No, you can't do this.

Your best workaround today is:
1. Make a cable between the SFP interface and the left-hand ODF port
2. Mark the right-hand ODF port as "connected" (to reserve it, and prevent any other cable being connected)

Alternatively, if your entire infrastructure uses only duplex connections, then you can model your ODFs where each connector represents a *pair* of fibres.

**Reply by subhashkoganti** [reply]
Thank you Brian for your response.

Yeah, we can pre-reserve alternate ports to address such situations.
Just that, we may have to accommodate both simplex and duplex connections.

We are still evaluating netbox.

Thanks again for your response

regards,
Subhash

**Comment by markkuleinio** [comment]
FWIW, in NetBox I label the fiber patch panel ports "1+2", "3+4", and so on, so that I can connect it to the device SFP port. And, when there is a simplex connection using one fiber only, I rename the patch panel port to use the one fiber number only (like "3") and then create the missing port ("4") to be used for another connection. Needs some fiddling but in my relatively static environment this works fine for now.

**Reply by subhashkoganti** [reply]
Thanks Markku.

I agree, that's a reasonable work-around as long as the functionality is used predominantly from UI.
At the same time, I think interpreting such data when used by other applications (via api) and reporting will become eventually painful.

**Comment by Torstein-Eide** [comment]
This how I got it to work:

### Manufacturer
add a Manufacturer, like `Generic`

### Device Type

- Select:  Manufacturer, like `Generic`
- type inn name of panel, Model.
- Other relevant data that you think is important. 

![image](https://github.com/netbox-community/netbox/assets/1884894/f875a881-660f-460a-a5ae-6eb048198a49)

**Save**

- Add a rear port
   - Name is up to you, 
   - Add type like `Splice`
   - Select a minimum 2 Positions (This is very importet). Think of it like strain in a fiber cabel, where a rear port is one cabel.
![image](https://github.com/netbox-community/netbox/assets/1884894/c7c18371-de75-4cdb-a9a6-6241f9074509)

- Add 2 front ports
   - Add name to ports
   - Select a type
   - Select the 2 Rear ports that is created

![image](https://github.com/netbox-community/netbox/assets/1884894/4d36f677-f257-4669-b0cd-874f78a7176d)


### Patch panel (PP)
add device: `.../dcim/devices/add/`

- Add 2 switch and 2 patchpanel to you test setup
![image](https://github.com/netbox-community/netbox/assets/1884894/3eb4cae8-823d-4e7b-8c58-16c137b92bf8)

1. From PP#1 rear to PP#2 add a rear connection.      
![image](https://github.com/netbox-community/netbox/assets/1884894/54b2131c-b4a9-4683-b1fa-8697401f6218)
![image](https://github.com/netbox-community/netbox/assets/1884894/4d6531e2-d10f-47d4-8672-68b11efd7e47)
2. From switch 1 to PP#1 add a connection
![image](https://github.com/netbox-community/netbox/assets/1884894/fdde9875-92b5-4575-a932-e4bd0c271bd5)      
Select both ports on the PP.
![image](https://github.com/netbox-community/netbox/assets/1884894/b74744ef-319d-4856-b2bc-6d32e86828b6)      

3.  From switch 2 to PP#2 add a connection.
![image](https://github.com/netbox-community/netbox/assets/1884894/fdde9875-92b5-4575-a932-e4bd0c271bd5)
Select both ports on the PP.
![image](https://github.com/netbox-community/netbox/assets/1884894/1b9df1d3-4605-46fc-9673-c9481da2c1d9)
      
4. You can now press `Trace` and you get the following:
![download](https://github.com/netbox-community/netbox/assets/1884894/53d0c366-d942-4d7e-b245-52f4d4d24482)

**Reply by candlerb** [reply]
Correct: you are answering an old thread. The ability to have a cable termination connect to multiple components was added in Netbox [v3.3.0](https://github.com/netbox-community/netbox/releases/v3.3.0) (Aug 17, 2022)
